### PR TITLE
Fixes #7383: Improve Rule compliance details loading

### PR DESCRIPTION
**Pull request by VinceMacBuche**
https://www.rudder-project.org/redmine/issues/7383

### CODE REVIEW

File: rudder-web/src/main/scala/com/normation/rudder/web/snippet/HomePage.scala [hunk 1]
```diff
@@ -332,6 +329,8 @@ class HomePage extends Loggable {
       n2               =  System.currentTimeMillis
       agentSoftEntries =  con.searchOne(acceptedNodesDit.SOFTWARE.dn, EQ(A_NAME, "rudder-agent"))
       agentSoftDn      =  agentSoftEntries.map(_.dn.toString).toSet
+      // Skip if there is no rudder-agent packages in software DN
+      if (! agentSoftDn.isEmpty)
```
Comment (fanf) [inline rudder-web/src/main/scala/com/normation/rudder/web/snippet/HomePage.scala]: With that, you are returning Empty, which was forbiden in the Tablet Of Law for "wtf is the semantic of Empy in an Error handling, Failable sentence ?".
Please just return Full(Map()) in that case, which should be the case without the if, I think (agentSoftEntries == Seq(), agentSoftDn == Set(), agentSoft == Seq(), perhaps just add a if(agentSoftDn.isEmpty) Seq() else { ... } for the nodeEntries part).

File: rudder-web/src/main/scala/com/normation/rudder/web/components/RuleEditForm.scala [hunk 2]
```diff
@@ -228,6 +226,7 @@ class RuleEditForm(
             | $$("#editRuleZone").bind( "show", function(event, ui) {
             | if(ui.panel.id== 'ruleComplianceTab') { ${ruleComplianceTabAjax}; }
             | });
+            |${Replace("details", new RuleCompliance(rule,directiveLib, nodeInfos, rootRuleCategory).display).toJsCmd};
```
Comment (fanf) [inline rudder-web/src/main/scala/com/normation/rudder/web/components/RuleEditForm.scala]: OK, that's nice. Moreover, if it takes too long, it won't block the "update form" part, and let user do stuff in the meantime.

**Comment by VinceMacBuche** [comment]
PR updated

**Comment by VinceMacBuche** [comment]
PR updated

**Comment by VinceMacBuche** [comment]
PR updated

**Pull request merged**